.github/workflows/release.yml

```diff
@@ -30,13 +30,19 @@ jobs:
       append_notes: ${{ steps.check_commits.outputs.append_notes }}
```

Comment:
1. The changes include adding a condition to check if the current workflow run is a rerun.
2. The code now fetches a list of workflow runs and identifies the last run that is not the current one.
3. Depending on whether it is a rerun or not, the code either skips tagging if the last run was successful or proceeds with tagging if it is a fresh run.
4. The comments have been updated to reflect the logic being implemented.
5. Overall, the changes add rerun detection and logic to handle tagging based on the status of the last run.